src/lib/gossip_net/libp2p.ml

```diff
@@ -462,14 +462,58 @@ module Make (Rpc_intf : Network_peer.Rpc_intf.Rpc_interface_intf) :
                 ~fn:(fun (env, vc) ->
                   match Envelope.Incoming.data env with
                   | Message.Latest.T.New_state state ->
-                      Sinks.Block_sink.push sink_block
-                        ( `Transition
-                            (Envelope.Incoming.map ~f:(const state) env)
-                        , `Time_received (Block_time.now config.time_controller)
-                        , `Valid_cb vc )
+                      let transactions =
```

Comment:
These new checks should be put into the network sinks, not this code. The network sinks are where we are applying the rest of the initial validation logic for gossip messages.